static/app/components/onboardingWizard/newSidebar.tsx

```diff
@@ -365,14 +372,88 @@ function Task({task, hidePanel, showWaitingIndicator}: TaskProps) {
       />
       {showSkipConfirmation && (
         <SkipConfirmation
-          onConfirm={() => handleMarkSkipped(task.task)}
+          onConfirm={() => {
+            handleMarkSkipped(task.task);
+            setShowSkipConfirmation(false);
+          }}
           onDismiss={() => setShowSkipConfirmation(false)}
         />
       )}
     </TaskWrapper>
   );
 }
 
+interface ExpandedTaskGroupProps {
+  hidePanel: () => void;
+  sortedTasks: OnboardingTask[];
+  taskKeyForWaitingIndicator: OnboardingTaskKey | undefined;
+}
+
+function ExpandedTaskGroup({
+  sortedTasks,
+  hidePanel,
+  taskKeyForWaitingIndicator,
+}: ExpandedTaskGroupProps) {
+  const api = useApi();
+  const organization = useOrganization();
+
+  const markCompletionTimeout = useRef<number | undefined>();
+
+  function completionTimeout(time: number): Promise<void> {
+    window.clearTimeout(markCompletionTimeout.current);
+    return new Promise(resolve => {
+      markCompletionTimeout.current = window.setTimeout(resolve, time);
+    });
+  }
+
+  const markTasksAsSeen = useCallback(
+    function () {
+      const unseenTasks = sortedTasks
+        .filter(task => taskIsDone(task) && !task.completionSeen)
+        .map(task => task.task);
+
+      for (const task of unseenTasks) {
+        updateOnboardingTask(api, organization, {task, completionSeen: true});
```

Comment:
is there no way to send only 1 api call?

Reply:
no we have to update the backend